### ISSUE TITLE
4.0a6:  runwsgi b0rken on Python3

### ISSUE THREAD

**Issue by tseaver** [zopefoundation/Zope#146]
```bash
$ /opt/Python-3.6.0/bin/virtualenv /tmp/z2py36
...
$ cd /tmp/z2py36
$ bin/pip install -U setuptools pip wheel
...
$ bin/pip install -r http://zopefoundation.github.io/Zope/releases/4.0a6/requirements-full.txt
...
$ bin/mkwsgiinstance -d . -u $LOGIN:$PASSWD
$ bin/runwsgi -v etc/zope.ini
```

provokes the following traceback:

```python
Traceback (most recent call last):
  File "bin/runwsgi", line 11, in <module>
    sys.exit(main())
  File "/tmp/z2py36/lib/python3.6/site-packages/Zope2/Startup/serve.py", line 215, in main
    return command.run()
  File "/tmp/z2py36/lib/python3.6/site-packages/Zope2/Startup/serve.py", line 182, in run
    global_conf=vars)
  File "/tmp/z2py36/lib/python3.6/site-packages/Zope2/Startup/serve.py", line 206, in loadapp
    return loadapp(app_spec, name=name, relative_to=relative_to, **kw)
  File "/tmp/z2py36/lib/python3.6/site-packages/paste/deploy/loadwsgi.py", line 247, in loadapp
    return loadobj(APP, uri, name=name, **kw)
  File "/tmp/z2py36/lib/python3.6/site-packages/paste/deploy/loadwsgi.py", line 272, in loadobj
    return context.create()
  File "/tmp/z2py36/lib/python3.6/site-packages/paste/deploy/loadwsgi.py", line 710, in create
    return self.object_type.invoke(self)
  File "/tmp/z2py36/lib/python3.6/site-packages/paste/deploy/loadwsgi.py", line 203, in invoke
    app = context.app_context.create()
  File "/tmp/z2py36/lib/python3.6/site-packages/paste/deploy/loadwsgi.py", line 710, in create
    return self.object_type.invoke(self)
  File "/tmp/z2py36/lib/python3.6/site-packages/paste/deploy/loadwsgi.py", line 146, in invoke
    return fix_call(context.object, context.global_conf, **context.local_conf)
  File "/tmp/z2py36/lib/python3.6/site-packages/paste/deploy/util.py", line 55, in fix_call
    val = callable(*args, **kw)
  File "/tmp/z2py36/lib/python3.6/site-packages/Zope2/Startup/run.py", line 58, in make_wsgi_app
    from ZPublisher.WSGIPublisher import publish_module
  File "/tmp/z2py36/lib/python3.6/site-packages/ZPublisher/WSGIPublisher.py", line 38, in <module>
    from ZPublisher.HTTPRequest import WSGIRequest
  File "/tmp/z2py36/lib/python3.6/site-packages/ZPublisher/HTTPRequest.py", line 48, in <module>
    from ZPublisher.BaseRequest import BaseRequest
  File "/tmp/z2py36/lib/python3.6/site-packages/ZPublisher/BaseRequest.py", line 51, in <module>
    from ZServer.ZPublisher.xmlrpc import is_xmlrpc_response
  File "/tmp/z2py36/lib/python3.6/site-packages/ZServer/__init__.py", line 71, in <module>
    utils.patchSyslogServiceName()
  File "/tmp/z2py36/lib/python3.6/site-packages/ZServer/utils.py", line 72, in patchSyslogServiceName
    from medusa import logger
ModuleNotFoundError: No module named 'medusa'
```

**Comment by hannosch** [comment]
Indeed. The issue here is that `ZServer==4.0a2` is in the requirements file. And since ZServer isn't Python 3 compatible at all, it fails with errors (relative vs. absolute imports in this case).

**Comment by tseaver** [comment]
The following fixes that error:

```bash
$ bin/pip uninstall -y ZServer
```

And then fails for `Products.MailHost`:

```bash
$ bin/runwsgi -v etc/zope.ini
```

traceback:

```python
$ bin/runwsgi -v etc/zope.ini 
Traceback (most recent call last):
  File "bin/runwsgi", line 11, in <module>
    sys.exit(main())
  File "/tmp/z2py36/lib/python3.6/site-packages/Zope2/Startup/serve.py", line 215, in main
    return command.run()
  File "/tmp/z2py36/lib/python3.6/site-packages/Zope2/Startup/serve.py", line 182, in run
    global_conf=vars)
  File "/tmp/z2py36/lib/python3.6/site-packages/Zope2/Startup/serve.py", line 206, in loadapp
    return loadapp(app_spec, name=name, relative_to=relative_to, **kw)
  File "/tmp/z2py36/lib/python3.6/site-packages/paste/deploy/loadwsgi.py", line 247, in loadapp
    return loadobj(APP, uri, name=name, **kw)
  File "/tmp/z2py36/lib/python3.6/site-packages/paste/deploy/loadwsgi.py", line 272, in loadobj
    return context.create()
  File "/tmp/z2py36/lib/python3.6/site-packages/paste/deploy/loadwsgi.py", line 710, in create
    return self.object_type.invoke(self)
  File "/tmp/z2py36/lib/python3.6/site-packages/paste/deploy/loadwsgi.py", line 203, in invoke
    app = context.app_context.create()
  File "/tmp/z2py36/lib/python3.6/site-packages/paste/deploy/loadwsgi.py", line 710, in create
    return self.object_type.invoke(self)
  File "/tmp/z2py36/lib/python3.6/site-packages/paste/deploy/loadwsgi.py", line 146, in invoke
    return fix_call(context.object, context.global_conf, **context.local_conf)
  File "/tmp/z2py36/lib/python3.6/site-packages/paste/deploy/util.py", line 55, in fix_call
    val = callable(*args, **kw)
  File "/tmp/z2py36/lib/python3.6/site-packages/Zope2/Startup/run.py", line 67, in make_wsgi_app
    starter.prepare()
  File "/tmp/z2py36/lib/python3.6/site-packages/Zope2/Startup/starter.py", line 38, in prepare
    self.startZope()
  File "/tmp/z2py36/lib/python3.6/site-packages/Zope2/Startup/starter.py", line 105, in startZope
    Zope2.startup_wsgi()
  File "/tmp/z2py36/lib/python3.6/site-packages/Zope2/__init__.py", line 49, in startup_wsgi
    _startup()
  File "/tmp/z2py36/lib/python3.6/site-packages/Zope2/App/startup.py", line 86, in startup
    OFS.Application.import_products()
  File "/tmp/z2py36/lib/python3.6/site-packages/OFS/Application.py", line 347, in import_products
    import_product(product_dir, product_name)
  File "/tmp/z2py36/lib/python3.6/site-packages/OFS/Application.py", line 357, in import_product
    global_dict, global_dict, ('__doc__', ))
  File "/tmp/z2py36/lib/python3.6/site-packages/Products/MailHost/__init__.py", line 14, in <module>
    import MailHost
ModuleNotFoundError: No module named 'MailHost'
```

Which then brings up the server:

```bash
$ bin/runwsgi -v etc/zope.ini 
2017-06-06 13:15:46 - WARNI [Init] Class Products.Five.browser.metaconfigure.+ has a security declaration for nonexistent method 'action.html'
Starting server in PID 21722.
Serving on http://localhost:8080
```

ISTM we need to be publishing a `requirements.txt` file which doesn't pull in the optional Python2-only deps.

**Comment by hannosch** [comment]
Should be fixed via 65a2bed071f4f7c147c7201d47414c15b055ec4f, by adding python_version markers to the requirements file.

**Comment by tseaver** [comment]
I'm going to claim that 65a2bed fixes this issue.
